webapp/lib/model.g.dart

```diff
@@ -77,64 +77,51 @@ class Conversation {
     };
   }
 
-  /// Add [tagId] to tagIds in this Conversation.
+  /// Add [newTagIds] to tagIds in this Conversation.
   /// Callers should catch and handle IOException.
-  Future<void> addTagId(DocPubSubUpdate pubSubClient, String tagId) {
-    return addTagIdToAll(pubSubClient, [this], tagId);
-  }
-
-  /// Add [tagId] to tagIds in each Conversation.
-  /// Callers should catch and handle IOException.
-  static Future<void> addTagIdToAll(DocPubSubUpdate pubSubClient, List<Conversation> docs, String tagId) async {
-    final docIdsToPublish = <String>[];
-    for (var doc in docs) {
-      if (!doc.tagIds.contains(tagId)) {
-        doc.tagIds.add(tagId);
-        docIdsToPublish.add(doc.docId);
+  Future<void> addTagIds(DocPubSubUpdate pubSubClient, Iterable<String> newTagIds) {
+    var toBeAdded = Set<String>();
+    for (var elem in newTagIds) {
+      if (!tagIds.contains(elem)) {
+        toBeAdded.add(elem);
       }
     }
-    if (docIdsToPublish.isEmpty) return;
-    return pubSubClient.publishDocAdd(collectionName, docIdsToPublish, {"tags": [tagId]});
+    if (toBeAdded.isEmpty) return Future.value(null);
+    tagIds.addAll(toBeAdded);
+    return pubSubClient.publishAddOpinion('nook_conversations/add_tags', {
+      'conversation_id': docId,
+      'tagIds': toBeAdded.toList(),
```

Comment:
@lukechurch Should this be `tagIds` or `tags` here?

Reply:
Sorry - let's call it 'tags' to be consistent with nook_messages/add_tags

I'll think about how to fix this in Piety so it can accept both, and we can standardise this.

Reply:
@lukechurch Sounds good. Fixed here and in set and remove methods